transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollSocketChannel.java

```diff
@@ -198,22 +194,46 @@ protected AbstractEpollUnsafe newUnsafe() {
         return new EpollSocketChannelUnsafe();
     }
 
+    private InetSocketAddress computeRemoteAddr(InetSocketAddress remoteAddr) {
+        InetSocketAddress osRemoteAddr = fd().remoteAddress();
```

Comment:
consider making this method static and take 2 parameters `private static InetSocketAddress computeRemoteAddr(InetSocketAddress requested, InetSocketAddress resolved)`. This may clarify the purpose and also avoid calling jni `fd().remoteAddress()` more than necessary.